### PR TITLE
fix: add support for Valkey

### PR DESCRIPTION
**Pull request by natoscott**
Enhancement: Support Valkey as an alternative to Redis

Reason: Redis is no longer open source and is being removed from distributions

Result: PCP scalable query language + Grafana support continue to function

Issue Tracker Tickets (Jira or BZ if any): RHEL-29939

### CODE REVIEW

File: tasks/firewall.yml [hunk 1]
```diff
@@ -30,12 +30,21 @@
       when:
         - metrics_graph_service|bool
 
+    - name: Service for valkey
+      set_fact:
+        __metrics_firewall: "{{ __metrics_firewall |
+            union([{'service': 'valkey', 'state': 'enabled'}]) }}"
+      when:
+        - metrics_query_service|bool
```
Comment (spetrosi) [inline tasks/firewall.yml:38]:
```suggestion
        - metrics_query_service | bool
```

File: tests/check_default_datasources.yml [hunk 2]
```diff
@@ -14,4 +14,25 @@
     - Redis
     - Vector
     - bpftrace
+  when: (ansible_distribution in ['RedHat', 'CentOS'] and
```
Comment (spetrosi) [inline tests/check_default_datasources.yml:17]: These three separate conditions can be written as a list
```
  when:
    - ansible_distribution in ['RedHat', 'CentOS']
    - ansible_distribution_major_version | int > 6
    - ansible_distribution_major_version | int < 10

File: tests/check_keyserver.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Check if Redis responds
+  shell: |
+    set -euo pipefail
+    redis-cli PING | grep PONG
+  changed_when: false
+  when: (ansible_distribution in ['RedHat', 'CentOS'] and
```
Comment (spetrosi) [inline tests/check_keyserver.yml:8]: convert to a list

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
@natoscott there is an error `chown failed: failed to look up user redis` in ci tests, please take a look

File: tasks/firewall.yml [hunk 4]
```diff
@@ -21,7 +21,7 @@
         __metrics_firewall: "{{ __metrics_firewall |
             union([{'port': '44322/tcp', 'state': 'enabled'}]) }}"
       when:
-        - metrics_graph_service|bool or metrics_query_service|bool
+        - metrics_graph_service | bool or metrics_query_service|bool
```
Comment (richm) [inline tasks/firewall.yml:24]:
```suggestion
        - metrics_graph_service | bool or metrics_query_service | bool
```

File: tasks/firewall.yml [hunk 5]
```diff
@@ -30,12 +30,21 @@
       when:
         - metrics_graph_service|bool
 
+    - name: Service for valkey
+      set_fact:
+        __metrics_firewall: "{{ __metrics_firewall |
+            union([{'service': 'valkey', 'state': 'enabled'}]) }}"
```
Comment (richm) [inline tasks/firewall.yml:36]: I don't see a `valkey` service:
```shell
firewall-cmd --get-services | grep valkey
```

File: tests/check_default_datasources.yml [hunk 6]
```diff
@@ -14,4 +14,26 @@
...
     - Vector
     - bpftrace
+  when:
+    - ansible_distribution in ['RedHat', 'CentOS']
+    - ansible_distribution_major_version | int > 6
+    - ansible_distribution_major_version | int < 10
+
+- name: Check if grafana-pcp datasources are configured
+  uri:
+    url: "http://admin:admin@localhost:3000/api/datasources/name/PCP%20{{ item }}"
```
Comment (richm) [inline tests/check_default_datasources.yml:24]: This seems unrelated?  and has a hardcoded password?

**Comment by natoscott** [comment]
> @natoscott there is an error `chown failed: failed to look up user redis` in ci tests, please take a look

Will do.  I have limited time currently, with CVEs, escalations and PTO - but I'll keep digging into this one as time is available (wont be this week) - thanks for all the reviews.

**Comment by natoscott** [comment]
[citest]

File: tasks/firewall.yml (hunk 5 above)
Comment (natoscott) [reply tasks/firewall.yml:36]: Hmm, this sounds like we'll need a bug opened in firewall-cmd to add valkey support - is that right?

File: tests/check_default_datasources.yml (hunk 6 above)
Comment (natoscott) [reply tests/check_default_datasources.yml:24]: Not sure why this is showing up as changed - will investigate (it is indeed unrelated & I didn't make this change anytime recently).  admin/admin is the default universally-known grafana password (since the dawn of time) - its always available after a new install IIRC.

Comment (natoscott) [reply tests/check_default_datasources.yml:24]: Oh, I think I've just sync'd this test up with the ansible-pcp equivalent check.

**Comment by natoscott** [comment]
[citest]

**Comment by natoscott** [comment]
[citest]

File: tasks/firewall.yml (hunk 5 above)
Comment (richm) [reply tasks/firewall.yml:36]: > Hmm, this sounds like we'll need a bug opened in firewall-cmd to add valkey support - is that right?

Yes, eventually, but in the meantime:
* use the port number directly e.g. `{'ports': 5678, 'state': 'enabled'}` (or whatever is the correct port)
* continue to use `redis` as the service, assuming that `valkey` is and will be identical for the foreseeable future

**Comment by natoscott** [comment]
[citest]

File: tasks/firewall.yml (hunk 5 above)
Comment (natoscott) [reply tasks/firewall.yml]: Done, went with option 1.

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
@natoscott I fixed the CI-related issue with ansible not being able to find `fedora.linux_system_roles.firewall` in https://github.com/linux-system-roles/tft-tests/pull/55, now there are some actual errors in CI run.

**Comment by richm** [comment]
```
TASK [fedora.linux_system_roles.private_metrics_subrole_keyserver : Ensure key server is configured] ***
task path: /tmp/collections-EIz/ansible_collections/fedora/linux_system_roles/roles/private_metrics_subrole_keyserver/tasks/main.yml:46
Monday 14 October 2024  10:23:34 -0400 (0:00:00.560)       0:01:02.532 ******** 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'redis_save_to_disk' is undefined
failed: [managed-node2] (item=/tmp/collections-EIz/ansible_collections/fedora/linux_system_roles/roles/private_metrics_subrole_keyserver/templates/CentOS_9_keyserver.conf.j2) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "/tmp/collections-EIz/ansible_collections/fedora/linux_system_roles/roles/private_metrics_subrole_keyserver/templates/CentOS_9_keyserver.conf.j2"
}

MSG:

AnsibleUndefinedVariable: 'redis_save_to_disk' is undefined
```

**Comment by natoscott** [comment]
[citest]

**Comment by natoscott** [comment]
OK, the only remaining CI failure I can see now is "AnsibleUndefinedVariable: 'performancecopilot' is undefined" which is unrelated to this change.  I think that might be related to ansible-pcp commit 9e4b26dbf66a which adds variables like "performancecopilot:ansible-pcp" but I'm not sure.
